pkg/cloudprovider/aws/fleet/packing/packing.go

```diff
@@ -77,34 +79,46 @@ func (p *podPacker) Pack(ctx context.Context, pods []*v1.Pod) ([]*Packing, error
 }
 
 // TODO filter instance types based on node contraints like availability zones etc.
-func (p *podPacker) getNodeCapacities() []*nodeCapacity {
+func (p *podPacker) getNodeCapacities(constraints *cloudprovider.Constraints) []*nodeCapacity {
+	p.reserveOnce.Do(
```

Comment:
I can see how this works but is it idiomatic Go when there isn't (IIUC?) the potential for concurrent calls to this code?

Reply:
Yes, there are no concurrent calls but we need to reserve overhead only once and we call `getNodeCapacities` every time we get pending pods.